### DISCUSSION TITLE
Migration guide for v2 -> v3?

### DISCUSSION THREAD

**Discussion by andymerskin** [ben-rogerson/twin.macro#778] [Ideas]
Hey there! I'm putting some notes together to prepare my team for migrating from Twin v2 to v3. The release notes scattered about for the v3 pre-releases and official release have a few small details mentioned for potential breaking changes, but don't ultimately give us a clear path for easy migration.

It would also be great to know what parts of the [Tailwind v3 Upgrade Guide](https://tailwindcss.com/docs/upgrade-guide) apply going from Twin v2 -> v3, with the obvious changes likely revolving around the TW configuration file.

---

This could be one of those things where we try to upgrade the package(s) and see what's working / isn't working, but I think for long-time v2 users who are hoping to benefit from Tailwind 3's amazing new features would greatly benefit from having a migration guide for Twin here.

If it's a pretty seamless upgrade, despite it being a major version bump, maybe a guide isn't needed — but at the very least, a mention in the README or docs somewhere to say it's seamless would be helpful for returning users.

Thanks for the great work in supporting Tailwind v3, I'm really looking forward to using it!

**Comment by ben-rogerson** [comment]
Hey there!
I don't think a guide is needed for this - I recently migrated a huge codebase with only a small number of changes required.
I avoided the short css updates for now with `{ disableShortCss: false }` in my twin config.

Here's the only resources you should need:

- [Tailwind v3 Upgrade Guide](https://tailwindcss.com/docs/upgrade-guide)
- [Twin release notes for v3](https://github.com/ben-rogerson/twin.macro/releases/tag/3.0.0)

**Reply by andymerskin** [reply]
Okay, great! Thanks for summarizing. Hopefully we will make the move soon. Tailwind's migration guide doesn't seem very daunting at all.
